File: java/planetapp/app/src/main/java/com/example/planetapp/MainActivity.java
package com.example.planetapp;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ListView listView;
    ArrayList<Planet> planetArrayList;
    private static MyCustomAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        listView = findViewById(R.id.listView);
        planetArrayList = new ArrayList<>();


        Planet planet1 = new Planet("Earth", "1", R.drawable.earth);
        Planet planet2 = new Planet("Mars", "2", R.drawable.mars);
        Planet planet3 = new Planet("Jupiter", "79", R.drawable.jupiter);
        Planet planet4 = new Planet("Saturn", "82", R.drawable.saturn);
        Planet planet5 = new Planet("Uranus", "27", R.drawable.uranus);
        Planet planet6 = new Planet("Sun", "0", R.drawable.sun);
        Planet planet7 = new Planet("Mercury", "0", R.drawable.mercury);
        Planet planet8 = new Planet("Venus", "0", R.drawable.venus);

        planetArrayList.add(planet1);
        planetArrayList.add(planet2);
        planetArrayList.add(planet3);
        planetArrayList.add(planet4);
        planetArrayList.add(planet5);
        planetArrayList.add(planet6);
        planetArrayList.add(planet7);
        planetArrayList.add(planet8);

        adapter = new MyCustomAdapter(planetArrayList, getApplicationContext());
        listView.setAdapter(adapter);



        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long id) {
                Toast.makeText(MainActivity.this,
                        "Planet name " + adapter.getItem(position).getPlanetName(), Toast.LENGTH_SHORT).show();
            }
        });












    }
}

File: kotlin/PlanetApp/app/src/main/java/com/example/planetapp/MainActivity.kt
package com.example.planetapp

import android.os.Bundle
import android.widget.ListView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var listview:ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        listview = findViewById(R.id.listView)

        val planets = Planets("Mercury", "0", R.drawable.mercury)
        val planets1 = Planets("Venus", "0", R.drawable.venus)
        val planets2 = Planets("Earth", "1", R.drawable.earth)
        val planets3 = Planets("Mars", "2", R.drawable.mars)
        val planets4 = Planets("Jupiter", "79", R.drawable.jupiter)
        val planets5 = Planets("Saturn", "82", R.drawable.saturn)
        val planets6 = Planets("Uranus", "27", R.drawable.uranus)
        val planets7 = Planets("Sun", "0", R.drawable.sun)

       var planetsList = ArrayList<Planets>()
        planetsList.add(planets)
        planetsList.add(planets1)
        planetsList.add(planets2)
        planetsList.add(planets3)
        planetsList.add(planets4)
        planetsList.add(planets5)
        planetsList.add(planets6)
        planetsList.add(planets7)

        listview.adapter = CustomAdapter(this, planetsList)


//        click handler
        listview.setOnItemClickListener { parent, view, position, id ->
            val item = parent.getItemAtPosition(position) as Planets
//          display the planet name in toast
            Toast.makeText(this, item.title, Toast.LENGTH_SHORT).show()
        }

    }
}